app/gui/src/project-view/components/FunctionSignatureEditor.vue

```diff
@@ -0,0 +1,47 @@
+<script setup lang="ts">
+import { WidgetInput } from '@/providers/widgetRegistry'
+import { computed, ref } from 'vue'
+import { FunctionDef } from 'ydoc-shared/ast'
+import WidgetTreeRoot from './GraphEditor/WidgetTreeRoot.vue'
+
+const { functionAst } = defineProps<{
+  functionAst: FunctionDef
+}>()
+
+const treeRootInput = computed(() => {
+  return WidgetInput.FromAst(functionAst)
+})
+
+const rootElement = ref<HTMLElement>()
+
+function handleWidgetUpdates() {
+  return true
+}
+</script>
+
+<template>
+  <div ref="rootElement" class="FunctionSignatureEditor define-node-colors">
+    <WidgetTreeRoot
+      :externalId="functionAst.externalId"
+      :input="treeRootInput"
+      :rootElement="rootElement"
+      :extended="true"
+      :onUpdate="handleWidgetUpdates"
+    />
+  </div>
+</template>
+
+<style scoped>
+.FunctionSignatureEditor {
+  margin: 4px 8px;
+  padding: 4px;
+
+  /* TODO */
```

Comment:
Oh. I completely forgot about it. I'm not sure what to do about it though at this point, since we don't have a known output type that would be used to color the node. I will just expand the comment here for now, explaining the issue.

Reply:
Actually ended up looking for defined group or output node's color, if available. I think this will be complete enough for current implementation.